### ISSUE TITLE
Compile-time error due to use of 3D syntax in debug-repl

### ISSUE THREAD

**Issue by lexi-lambda** [AlexKnauth/debug#10]
Compiling `debug` currently fails due to [the use of 3D syntax in `debug/repl.rkt`](https://github.com/AlexKnauth/debug/blob/0424641fce57efb915d12e6576b8c87aef7a826a/debug/repl.rkt#L42-L46). I see the purpose of this, and it does appear to work as long as the code isn’t compiled, but it causes `raco setup` errors, which is annoying. I’m not totally sure how to work around it, though, since this sort of intentionally passes a value between phase levels in a way that the compiler can’t understand.

**Comment by AlexKnauth** [comment]
Yes. I *am* trying to preserve macro transformer values so I can use them at run-time in a Repl. My current solution works in situations where I run and compile at the same time in DrRacket, but I need a better solution that also works when the code is compiled.

**Comment by lexi-lambda** [comment]
I’m not sure that it’s (currently) possible. Plus, as you mentioned in the commit message, it only works in simple cases where macros do not use other macros. This seems like something that might need support from the compiler or the expander? It would be useful to have a robust solution for this that works in 100% of cases; this sort of feature is extremely useful for debugging. Ruby’s `pry` library provides a `binding.pry` that does the same thing (opens a REPL at the location at runtime), and it provides one of the nicest debugging workflows I’ve ever used.

**Comment by AlexKnauth** [comment]
I've split those tests into a separate file which I can add to `compile-omit-paths`. When I do this it compiles and runs at the same time so the tests pass. This is not a solution, but it means the docs can build.
